app/decorators/course_decorator.rb

```diff
@@ -130,6 +130,27 @@ def sorted_subjects
     object.subjects.map(&:subject_name).sort.join("<br>").html_safe
   end
 
+  def chosen_subjects(course)
```

Comment:
This method is handling quite a few things. I think we could break it up into something like:

```ruby
  def chosen_subjects(course)
    return sorted_subjects if course.master_subject_id.nil?

    return format_name(additional_subjects.push(main_subject).reverse) if main_subject?

    format_name(additional_subjects.unshift(main_subject))
  end

  def additional_subjects
    course.subjects.map { |subject| subject unless subject.id == main_subject.id }.compact
  end

  def main_subject
    Subject.find(course.master_subject_id)
  end

  def main_subject?
    main_subject.id == SecondarySubject.modern_languages.id
  end

  def format_name(subjects)
    subjects.map(&:subject_name).join("<br>").html_safe
  end

```

You may have to tweak and check your tests. Smaller methods are easier to grok and understand and having the `main_subject` as a little helper might help with other bits in this file.

Reply:
Nice! Now it looks like a professional wrote it 😆 

From 16 lines to 3. 💯